modules/core/test/test_quaternion.cpp

```diff
@@ -257,6 +257,183 @@ TEST_F(QuatTest, interpolation){
     EXPECT_EQ(Quatd::spline(tr1, tr2, tr3, tr3, 0.5), -Quatd::spline(-tr1, -tr2, -tr3, tr3, 0.5));
     EXPECT_EQ(Quatd::spline(tr1, tr2, tr3, tr3, 0.5), Quatd(0.336889853392, 0.543600719487, 0.543600719487, 0.543600719487));
 }
+TEST_F(QuatTest, EulerAngles){
+    Vec3d test_angle={0.523598,0.78539,1.04719};
+    Quatd qEuler0 = Quatd(0, 0, 0, 0);
+    Quatd qEuler1 = Quatd::createFromEulerAngles(test_angle, Quatd::INT_XYZ);
+    Quatd qEuler2 = Quatd::createFromEulerAngles(test_angle, Quatd::INT_XZY);
+    Quatd qEuler3 = Quatd::createFromEulerAngles(test_angle, Quatd::INT_XZX);
```

Comment:
Consider using arrays and for loops to process/check values. Like `qEuler[24]`
This should reduce amount of code.

Use error messages to help with problem investigations. See proposed commits.

Reply:
OK, we use for loops  and array `qEuler[24]` in test codes to reduce duplicate code.